src/main/java/org/openelisglobal/testconfiguration/controller/TestAddController.java

```diff
@@ -284,11 +285,11 @@ private ArrayList<ResultLimit> createResultLimits(Double lowValid, Double highVa
             limit.setHighNormal(StringUtil.doubleWithInfinity(params.highLimit));
             limit.setLowValid(lowValid);
             limit.setHighValid(highValid);
-            if(lowReportingRange != null && highReportingRange != null && highCriticalRange != null && lowCriticalRange != null){
-            limit.setLowReportingRange(lowReportingRange);
-            limit.setHighReportingRange(highReportingRange);
-            limit.setLowCriticalRange(lowCriticalRange);
-            limit.setHighCriticalRange(highCriticalRange);
+            if(limit != null)    {
```

Comment:
I think this null check is unnecessary and should be converted to the 3 independent null checks mentioned above (3 separate ifs).